src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs

```diff
@@ -635,9 +636,47 @@ private TestMethodAttribute GetTestMethodAttribute(MethodInfo methodInfo, TestCl
         /// <returns> The <see cref="MethodInfo"/>. </returns>
         private MethodInfo GetMethodInfoForTestMethod(TestMethod testMethod, TestClassInfo testClassInfo)
         {
-            var methodsInClass = testClassInfo.ClassType.GetRuntimeMethods().ToArray();
             MethodInfo testMethodInfo;
 
+            if (testMethod.HasManagedMethodAndType)
+            {
+                var assembly = testClassInfo.ClassType.GetTypeInfo().Assembly;
+                var methodBase = ManagedNameHelper.GetMethod(assembly, testMethod.ManagedTypeName, testMethod.ManagedMethodName);
```

Comment:
Do we handle function overloading today? What would happen if this is an overloaded function?

Reply:
ManagedMethodName includes the parameters as well. We’re correctly handling overloading.